pytorch_lightning/loops/optimization/optimizer_loop.py

```diff
@@ -52,36 +52,45 @@ def __init__(self) -> None:
         self._skip_backward: bool = False
         self._batch_idx: int = 0
         self._optimizers: List[Optimizer] = []
+        self._indices: List[int] = []
+        self._current_position = 0
         self._hiddens: Optional[Any] = None
 
+    @property
+    def optimizer_idx(self) -> int:
+        return self._indices[self._current_position]
+
     @property
     def done(self) -> bool:
         """Returns ``True`` when the last optimizer in the sequence has run."""
-        return self.optim_progress.optimizer_idx >= len(self._optimizers)
+        return self._current_position >= len(self._indices)
 
     def connect(self, **kwargs: "Loop") -> None:
         raise NotImplementedError(f"{self.__class__.__name__} does not connect any child loops.")
 
     def reset(self) -> None:
         if not self.restarting or self.done:
-            self.optim_progress.optimizer_idx = 0
```

Comment:
If the `optim_progress.optimizer_idx` is not used to restart anymore, that's saying to me that we don't need to save it in the dataclass.

Should we save `optimizer_idxs: List[int], current_position = 0` as part of the dataclass?

Reply:
we could do that, though saving the list of indices is redundant if we know the current position. 
I think a separate PR is needed to restore the optimizer loop properly

Reply:
still i'd rename `optimizer_idx` to `optimizer_position` and remove `self._current_position` from this class in this PR

Reply:
okay that's done